### PR TITLE
docs: Add guides for calico cleanup after upgrading to v3.5

### LINKED ISSUE #908: document calico 3.5 upgrade process
The tl;dr is that some manual cleanup will be required for users upgrading clusters that already have an aks-engine addon-manager-enforce calico spec.

Here's a gist that @ahrkrak provided as a reference for helping to develop a definitive howto doc:

https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgist.githubusercontent.com%2Fsong-jiang%2F5a8cdbd5b79ad379506d7e1b83c4685f%2Fraw%2F4f5f643fe844a92237e8f18fa3068cf0248bb31c%2Faks-engine-calico-3.3.1-delete-after-upgrade.yaml&data=02%7C01%7CJack.Francis%40microsoft.com%7Cad0c17af6dcd402e89b408d6aeed59f7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888731779430243&sdata=vcuBtL9xz3eh4WW1AXwBhhkxMNOsD6oZw%2BQ9DPRn6M0%3D&reserved=0

### PR DESCRIPTION
**Pull request by feiskyer**
**Reason for Change**:

Add guides for calico cleanup after upgrading to v3.5.

**Issue Fixed**:

Fixes #908.

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: docs/topics/features.md [hunk 1]
```diff
@@ -170,6 +170,16 @@ Per default Calico still allows all communication within the cluster. Using Kube
 - [NetworkPolicy Example Walkthrough](https://kubernetes.io/docs/getting-started-guides/network-policy/walkthrough/)
 - [Calico Kubernetes](https://github.com/Azure/aks-engine/blob/master/examples/networkpolicy)
 
+### Calico 3.3 cleanup after upgrading to 3.5
```
Comment (CecileRobertMichon) [inline docs/topics/features.md]: would https://github.com/Azure/aks-engine/blob/master/examples/networkpolicy/README.md be a better place to put this?

Comment (CecileRobertMichon) [reply docs/topics/features.md]: also see https://github.com/Azure/aks-engine/blob/master/examples/networkpolicy/README.md#update-guidance-for-clusters-deployed-by-aks-engine-releases-prior-to-0170

Comment (jackfrancis) [reply docs/topics/features.md]: Ha, I already suggested to Pengfei to move to docs/

But generally open to whatever makes the most sense.

File: docs/topics/features.md [hunk 2]
```diff
@@ -170,6 +170,25 @@ Per default Calico still allows all communication within the cluster. Using Kube
...
+
+Because Calico 3.3 is using Calico CNI, while Calico 3.5 moves to Azure CNI, if the cluster is upgraded from calico 3.3 to 3.5, then some manual cluster resource cleanup will be required to successfully complete the upgrade. We've provided a sample resource spec here that can be used as an example:
+
+https://github.com/Azure/aks-engine/raw/master/docs/topics/calico-3.3.1-cleanup-after-upgrade.yaml
+
+There are some placeholder tokens in the above `yaml` file, so please reconcile those with the actual values in your cluster. Look for these placeholder strings in the spec, then compare with the running spec of the comparable pre-3.5 resource in your cluster, and modify the cleanup spec accordingly:
+
```
Comment (feiskyer) [inline docs/topics/features.md]: Since the manifest is only used for cleanup, those changes are actually not required.

**Pull request merged**